webapp/channels/src/components/more_channels/more_channels.tsx

```diff
@@ -214,87 +250,87 @@ export default class MoreChannels extends React.PureComponent<Props, State> {
+                        aria-label={localizeMessage('more_channels.create', 'Create New Channel')}
+                    >
+                        {icon}
+                        <FormattedMessage
+                            id='more_channels.create'
+                            defaultMessage='Create New Channel'
+                        />
+                    </button>
+                </TeamPermissionGate>
+            );
+        };
 
-        const createChannelHelpText = (
-            <TeamPermissionGate
-                teamId={teamId}
-                permissions={[Permissions.CREATE_PUBLIC_CHANNEL, Permissions.CREATE_PRIVATE_CHANNEL]}
-            >
+        const noResultsText = (
+            <>
                 <p className='secondary-message'>
                     <FormattedMessage
-                        id='more_channels.createClick'
-                        defaultMessage="Click 'Create New Channel' to make a new one"
+                        id='more_channels.searchError'
+                        defaultMessage='Try searching different keywords, checking for typos or adjusting the filters.'
                     />
                 </p>
-            </TeamPermissionGate>
+                {createNewChannelButton('primaryButton', <i className='icon-plus'/>)}
+            </>
         );
 
-        const body = (
+        const body = this.state.loading ? <LoadingScreen/> : (
             <React.Fragment>
                 <SearchableChannelList
-                    channels={activeChannels}
+                    channels={this.activeChannels}
                     channelsPerPage={CHANNELS_PER_PAGE}
                     nextPage={this.nextPage}
                     isSearch={search}
                     search={this.search}
                     handleJoin={this.handleJoin}
-                    noResultsText={createChannelHelpText}
+                    noResultsText={noResultsText}
                     loading={search ? searching : channelsRequestStarted}
                     toggleArchivedChannels={this.toggleArchivedChannels}
                     shouldShowArchivedChannels={this.state.shouldShowArchivedChannels}
                     canShowArchivedChannels={this.props.canShowArchivedChannels}
+                    myChannelMemberships={this.props.myChannelMemberships}
+                    closeModal={this.props.actions.closeModal}
+                    hideJoinedChannelsPreference={this.handleShowJoinedChannelsPreference}
+                    rememberHideJoinedChannelsChecked={shouldHideJoinedChannels}
                 />
                 {serverError}
             </React.Fragment>
         );
 
+        const title = (
+            <FormattedMessage
+                id='more_channels.title'
+                defaultMessage='Browse Channels'
+            />
+        );
+
         return (
-            <Modal
-                dialogClassName='a11y__modal more-modal more-modal--action'
```

Comment:
I'm glad to see that we're getting rid of the original usage of `more-modal`. That's one of those overused classes in MM that makes working with our CSS styling difficult, so this'll definitely help to get rid of that eventually